### PR TITLE
Usman/byte collection updation

### PR DESCRIPTION
**Pull request by UsmanSaleem2003**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [x] I have carefully reviewed my code and believe it is final from my side.
- [x] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [x] I have tested the new feature or updates in both light and dark themes.
- [x] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [x] I have tested my changes on mobile screen sizes.
- [x] I have tested my changes on tablet screen sizes.
- [x] I have tested my changes on laptop screen sizes.

### Functional Testing
- [x] I have thoroughly tested the functionality of my changes.
- [x] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [x] Explain the task/bug you are solving
- [x] Show the changes in the UI
- [x] Show the changes in the codebase
- [x] Show the changes you tested to make sure it works
- [x] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

**Loom Video Link :** https://www.loom.com/share/1ed00b20a23b461892c2e9668cec43d8?sid=d36d800b-3738-4a1b-992e-f0b2044acddf

### CODE REVIEW

File: academy-ui/src/app/tidbit-collections/edit/[[...tidbitCollectionId]]/page.tsx [hunk 1]
```diff
@@ -135,7 +135,6 @@ function EditTidbitCollectionSpace(props: { space: SpaceWithIntegrationsFragment
             space={props.space}
             byteCollection={data!.byteCollection}
             viewByteCollectionsUrl={'/tidbit-collections'}
-            byteSummaries={bytesResponse?.bytes}
```
Comment (RobinNagpal) [inline academy-ui/src/app/tidbit-collections/edit/[[...tidbitCollectionId]]/page.tsx:138]: Can you please remove the old code also related to selection of the bytes in the modal. As that code is a dead code now

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx [hunk 2]
```diff
@@ -5,25 +5,67 @@ import {
...
+export default function ByteCollectionCardAdminDropdown({ byteCollection, byteCollectionType, project, space }: ByteCollectionCardAdminDropdownProps) {
   const { showNotification } = useNotificationContext();
   const [showDeleteModal, setShowDeleteModal] = React.useState<boolean>(false);
   const [editProjecByteCollectionSeo, setEditProjectByteCollectionSeo] = React.useState<boolean>(false);
+  const [showEditCollectionModal, setShowEditCollectionModal] = React.useState<boolean>(false);
+  const router = useRouter();
+
+  function onClose() {
+    setShowEditCollectionModal(false);
+    router.push(`/tidbit-collections`);
```
Comment (RobinNagpal) [inline academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:35]: Why do we need to reload or update the route?

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx [hunk 3]
```diff
@@ -5,25 +5,67 @@ import {
...
+    router.refresh();
```
Comment (RobinNagpal) [inline academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:36]: Why do we need router.refresh ?

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx [hunk 4]
```diff
@@ -5,25 +5,67 @@ import {
...
+  }
+
+  async function upsertByteCollectionFn(byteCollectionn: EditByteCollection) {
+    try {
```
Comment (RobinNagpal) [inline academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:40]: We should show some loading/updating icon when the byte collection is getting updated/added

File: academy-ui/src/components/byteCollection/View/ByteCollectionsGrid.tsx [hunk 5]
```diff
@@ -1,22 +1,72 @@
+'use client';
```
Comment (RobinNagpal) [inline academy-ui/src/components/byteCollection/View/ByteCollectionsGrid.tsx:1]: We should not use `use client` here. Add of the edition or updation code can move to a new component that has `use client`

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx (hunk 2 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:35]: > Why do we need to reload or update the route?

So, when we update a tidbit collection, changes doesn't reflect immediately. To make sure, that after updating, changes must reflect without user's manual reload, I am reloading the route here.

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx (hunk 3 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:36]: To make sure, the changes reflect immediately after add/update

File: academy-ui/src/app/tidbit-collections/edit/[[...tidbitCollectionId]]/page.tsx (hunk 1 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/app/tidbit-collections/edit/[[...tidbitCollectionId]]/page.tsx:138]: I have removed SelectBytesModal.tsx file which was used for selection of existing bytes to be added in a collection.

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx (hunk 3 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:36]: I am removed line :  router.push(`/tidbit-collections`). And just kept below line to immediately reflect the changes after updating/creating.

File: academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx (hunk 4 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/ByteCollections/ByteCollectionsCard/ByteCollectionCardAdminDropdown.tsx:40]: Now, showing loading icon on button when byte collection is getting updated/added.

File: academy-ui/src/components/byteCollection/View/ByteCollectionsGrid.tsx (hunk 5 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/View/ByteCollectionsGrid.tsx]: Created a new file AddByteCollection.tsx and moved necessary code to that component and called that component in ByteCollectionsGrid file.

File: academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import { useNotificationContext } from '@dodao/web-core/ui/contexts/NotificationContext';
+
+export default function AddByteCollection({ space }: { space: SpaceWithIntegrationsFragment }) {
+  const [showAddCollectionModal, setShowAddCollectionModal] = React.useState(false);
+  const { showNotification } = useNotificationContext();
+  const router = useRouter();
+
+  function onClose() {
+    setShowAddCollectionModal(false);
+    router.refresh();
```
Comment (RobinNagpal) [inline academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx:20]: Why do we need to do router.refresh() here?

Comment (RobinNagpal) [reply academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx:20]: It does a reload and the experience is not good

File: academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      }
+    } catch (error) {
+      showNotification({ message: 'Something went wrong', type: 'error' });
+    }
+  }
+
+  return (
+    <>
+      <div className="w-full flex items-center justify-center">
+        <button
```
Comment (RobinNagpal) [inline academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx:55]: We should use one of the existing button components

File: academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  const { showNotification } = useNotificationContext();
+  const router = useRouter();
+
+  function onClose() {
+    setShowAddCollectionModal(false);
+    router.refresh();
+  }
+
+  async function upsertByteCollectionFn(byteCollection: EditByteCollection) {
+    try {
```
[the remark this answers is not in the document]
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx:24]: It was implemented on the button. When you click on button for upsertion, it gets disabled and shows a circular wheel type animation (like it's loading).

File: academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx (hunk 6 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollection/ByteCollections/AddByteCollection.tsx:20]: Because, if we don't reload, the user will have to manually reload which isn't a good user experience. After upsertion, changes don't reflect until it's reloaded.
That's why I am reloading it.
